src/core/Modal/Modal.tsx

```diff
@@ -0,0 +1,57 @@
+import { Modal as KuberaModal, Button } from '@material-ui/core';
+import React from 'react';
+import useStyles from './styles';
+
+interface ModalProps {
+  isOpen: boolean;
+  handleClose: () => void;
+  hasCloseBtn: boolean;
+}
+
+const Modal: React.FC<ModalProps> = ({ children, hasCloseBtn }) => {
+  const classes = useStyles();
+  const [isOpen, setOpen] = React.useState(false);
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
```

Comment:
If the modal open and close is a local state then we loose functionality to manually control this programmatically from the website it is being used in. External APIs for open and handleClose needs to be exposed